File: backend/src/main/java/com/bomnuocv1/domain/repository/PaymentRepository.java
package com.bomnuocv1.domain.repository;

import com.bomnuocv1.domain.entity.Payment;

import java.math.BigDecimal;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface PaymentRepository {

    Payment save(Payment payment);

    Optional<Payment> findById(UUID id);

    List<Payment> findByOwnerId(UUID ownerId);

    List<Payment> findByTransactionId(UUID transactionId);

    List<Payment> findByFarmerId(UUID farmerId);

    BigDecimal sumPaidByTransactionId(UUID transactionId);
}

File: frontend/app/src/main/java/com/vn/bomnuocv1/domain/repository/PaymentRepository.kt
package com.vn.bomnuocv1.domain.repository

import com.vn.bomnuocv1.domain.model.Payment
import java.math.BigDecimal

data class CreatePaymentParams(
    val farmerId: String,
    val transactionId: String?,
    val amount: BigDecimal,
    val paymentDate: String,
    val note: String?
)

interface PaymentRepository {

    suspend fun recordPayment(
        params: CreatePaymentParams
    ): Result<Payment>

    suspend fun getPayments(
        farmerId: String? = null,
        transactionId: String? = null
    ): Result<List<Payment>>
}
